overlord/configstate/configcore/homedirs.go

```diff
@@ -117,6 +117,9 @@ func updateHomedirsConfig(config string, opts *fsOnlyContext) error {
 		return err
 	}
 
+	// Update value in dirs
+	dirs.SetSnapHomeDirs(config)
```

Comment:
small suggestion: if this returned a []string we wouldn't need the Split again below